sdk/Lusid.Sdk.Tests/Utilities/PollyApiRetryHandlerTests.cs

```diff
@@ -315,6 +317,91 @@ public void CreateLusidFactory_WhenRetryPolicyIsAlreadyAssigned_ExistingRetryPol
+            AddMockHttpResponseToQueue(_httpListener, statusCode: retryAfterResponseCode, responseContent: "", 
+                0, new Dictionary<HttpResponseHeader, string>() {[HttpResponseHeader.RetryAfter] = "1"});
+            
+            AddMockHttpResponseToQueue(_httpListener, statusCode: retryAfterResponseCode, responseContent: "", 
+                0, new Dictionary<HttpResponseHeader, string>() {[HttpResponseHeader.RetryAfter] = "1"});
+
+            AddMockHttpResponseToQueue(_httpListener, statusCode: statusCodeResponseDefaultRetry, responseContent: "");
+            
+            AddMockHttpResponseToQueue(_httpListener, statusCode: statusCodeResponseDefaultRetry, responseContent: "");
+
+            AddMockHttpResponseToQueue(_httpListener, statusCode: 200, responseContent: _testPortfolioResponse.ToJson());
+          
+            RetryConfiguration.RetryPolicy = PollyApiRetryHandler.DefaultRetryPolicyWithRateLimit;
+            
+            // Calling the API triggers the flow that triggers polly
+            var sdkResponse = _apiFactory.Api<IPortfoliosApi>().GetPortfolio("any","any");
+            
+            Assert.That(sdkResponse, Is.EqualTo(_testPortfolioResponse));
+            Assert.That(_apiCallCount, Is.EqualTo(expectedNumberOfApiCalls));
+        }
+
+        [Test]
+        public void UseRateLimitPolicyWithRetryAfter_WhenCallingApiMethodHitsRateLimit_RetryAfterIsHonored()
+        {
+            const int retryAfterResponseCode = 429;
+            const int expectedNumberOfApiCalls = 4; // 1 initial call, 2 failed retries and one success
+            
+            // First Response
+            AddMockHttpResponseToQueue(_httpListener, statusCode: retryAfterResponseCode, responseContent: "", 
+                0, new Dictionary<HttpResponseHeader, string>() { [HttpResponseHeader.RetryAfter] = "5" });
+            // Second Response - same, triggers another retry
+            AddMockHttpResponseToQueue(_httpListener, statusCode: retryAfterResponseCode, responseContent: "", 
+                0, new Dictionary<HttpResponseHeader, string>() { [HttpResponseHeader.RetryAfter] = "7" });
+            
+            AddMockHttpResponseToQueue(_httpListener, statusCode: retryAfterResponseCode, responseContent: "", 
+                0, new Dictionary<HttpResponseHeader, string>() { [HttpResponseHeader.RetryAfter] = "9" });
+            // 4 time lucky:
+            AddMockHttpResponseToQueue(_httpListener, statusCode: 200, responseContent: _testPortfolioResponse.ToJson());
+
+            RetryConfiguration.RetryPolicy = PollyApiRetryHandler.RateLimitRetryPolicy;
+            var sw = Stopwatch.StartNew();
+            // Calling the API triggers the flow that triggers polly
+            var sdkResponse = _apiFactory.Api<IPortfoliosApi>().GetPortfolio("any","any");
+            sw.Stop();
+            Assert.That(sw.ElapsedMilliseconds, Is.GreaterThanOrEqualTo(1000*21)); // retry after was respected
```

Comment:
just for clarity, is it with putting `5 + 7 + 9` rather than `21`, or add a comment explaining where `21`` came from?